core-rust/mesh-api-server/src/mesh_api/handlers/construction_parse.rs

```diff
@@ -0,0 +1,197 @@
+                            account: Some(Box::new(to_mesh_api_account_from_address(
+                                mapping_context,
+                                global_address,
+                            )?)),
+                            amount: Some(Box::new(to_mesh_api_amount(
+                                -input.amount.clone(),
+                                to_mesh_api_currency_from_resource_address(
+                                    mapping_context,
+                                    database,
+                                    &match input.resource_address {
+                                        ManifestResourceAddress::Static(resource_address) => {
+                                            resource_address
+                                        }
+                                        ManifestResourceAddress::Named(_) => {
+                                            return Err(ResponseError::from(
+                                                ApiError::NamedAddressNotSupported,
+                                            )
+                                            .with_details("Named address is not supported"))
+                                        }
+                                    },
+                                )?,
+                            )?)),
+                            coin_change: None,
+                            metadata: None,
+                        });
+                    }
+                    _ => {
+                        return Err(ResponseError::from(ApiError::UnrecognizedInstruction)
+                            .with_details(format!("Unrecognized instruction: {:?}", instruction)));
+                    }
+                }
+            }
+            InstructionV1::TakeFromWorktop(TakeFromWorktop {
+                resource_address,
+                amount,
+            }) if next_index < instructions.len() => {
+                instruction = &instructions[next_index];
+                next_index = next_index + 1;
+                if let InstructionV1::CallMethod(CallMethod {
+                    address: DynamicGlobalAddress::Static(global_address),
+                    method_name,
+                    args,
+                }) = instruction
+                {
+                    if method_name.eq("try_deposit_or_abort") {
+                        if let Ok(_input) = manifest_decode::<AccountTryDepositOrAbortManifestInput>(
+                            &manifest_encode(&args).unwrap(),
+                        ) {
+                            operations.push(Operation {
+                                operation_identifier: Box::new(OperationIdentifier {
+                                    index: operations.len() as i64,
+                                    network_index: None,
+                                }),
+                                related_operations: None,
+                                _type: "Deposit".to_owned(),
```

Comment:
We should use `MeshApiOperationType::Deposit.to_string()` here

Reply:
Done